workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Improbable.Gdk.Core;
+using Improbable.Gdk.Mobile;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Playground
+{
+    public class ConnectionScreenController : MonoBehaviour
+    {
+        private const string CachedIp = "cachedIp";
+
+        [SerializeField] private GameObject connectionPanel;
+        [SerializeField] private InputField ipAddressInput;
+        [SerializeField] private Button connectButton;
+        [SerializeField] private Text errorMessage;
+        [SerializeField] private GameObject clientWorkerPrefab;
+
+        private GameObject worker;
+
+        private string IpAddress => ipAddressInput != null ? ipAddressInput.text : null;
+
+        public void Awake()
+        {
+            ipAddressInput.text = PlayerPrefs.GetString(CachedIp);
+            connectButton.onClick.AddListener(TryConnect);
+        }
+
+        private void TryConnect()
+        {
+            worker = Instantiate(clientWorkerPrefab);
+            var workerConnector = worker.GetComponent<IMobileClient>();
+
+            if (workerConnector == null)
+            {
+                return;
```

Comment:
Is this an error that should be logged/thrown?

Reply:
Aye, or show something in the UI